docs/06-contributors/999-rfcs/000-stories/story-15.md

```diff
@@ -0,0 +1,173 @@
+- From anything to Json - can I cast something from bring untyped to Json? 
+- Calling await 
+
+## Code 
+```ts (wing)
+
+bring cloud;
+
+// TODO discuss how we bring untyped something like RegEx from JavaScript 
+// PLACEHOLER for bringing something from Javascript stdlib
+bring untyped RegExp from_js("RegExp"); 
+bring untyped Math from_js("Math"); 
+
+
+// prerequisite: npm install axios
+// PALCEHOLDER for bringing some external module
+bring untyped axios require_from_js("axios"); 
+
+enum Status {
+  Uncompleted,
+  Completed
+}
+
+interface ITaskListModel {
+  inflight get(id: str): Json;
+  inflight add(title: str): str;
+  inflight remove(id: str): void; 
+  inflight find(r: RegExp): Array<str>;
+  inflight set_status(id: str, status: Status): str;
+  inflight set_estimation(id: str, estimation: duration): str;
+}
+
+resource TaskListModel implementes ITaskListModel {
+  _bucket: cloud.Bucket;
+  init() {
+    this._bucket = new cloud.Bucket();
+    this.inflights.add("get", ref: "this._bucket", op: "get");
+    this.inflights.add("_add", ref: "this._bucket", op: "put");
+    // notice I am setting explicit permissions on this
+    this.inflights.add("add", ref: "this", op: "_add");
+    // is this the right synatx for multiple ops? 
+    this.inflights.add("set_status", ref: "this", op: ["_add", "get"]); 
+    this.inflights.add("set_estimation", ref: "this", op: ["_add", "get"]); 
+
+    // TODO add more permissions for remove, find
+  }
+
+  inflight get(id: str): Json {
+     return this._bucket.get_json(id);
+  }
+  
+  inflight _add(id: str, j: Json): str {
+    this._bucket.put_json(id, j);
+    return id;
+  } 
+  
+  inflight add(title: str): str {
+    // PLACEHOLDER - how does untyped works with numeric operations
+    let id = "${Math.floor(Math.random() * 100000000000)}";
+    let j = Json { 
+      title: title, 
+      status: Status.Uncompleted
+    };
+    print("adding task ${id} with data: ${j}"); 
+    return this_add(id, js);
+  }
+
+  inflight remove(id: str) {
+    print("removing task ${id}");
+    this._bucket.delete(id);
+    return id;
+  }
+
+  inflight find(r: RegExp): Array<str> { 
+    //TODO add implementation
+  }
+
+  inflight set_status(id: str, status: Status): str {
+    let j = Json.clone_mut(this.get(id));
+    j.status = status;
+    this._add(id, j);
+    return id;
+  }
+
+  inflight set_estimation(id: str, estimation: duration): str {
+    let j = Json.clone_mut(this.get(id));
+    j.effort_estimation = estimation;
+    this._add(id, j);
+    return id;
+  }
+}
+
+resource TaskListApi {
+  api: cloud.Api;
+  model: TaskListModel;
+  init(model: ITaskListModel) {
+    this.model = model;
+    this.api = new cloud.Api();
+    
+    // TODO add this.put
+    
+    this.api.post("/tasks", inflight (req: cloud. Api.ApiRequest): cloud.ApiResponse => {
+      let var title = str.from_json(req.body.title);
+      if title == random {
```

Comment:
what does that mean? how is it related to this story?

Reply:
I've added an easter egg  to our TODO list, if you add a title random, the system will fetch a random task from the Internet. 

We wanted to use network for something, and thought this is a nice idea